rclpy/rclpy/context.py

```diff
@@ -60,28 +60,29 @@ def init(self,
         from rclpy.impl.implementation_singleton import rclpy_implementation
 
         global g_logging_ref_count
-        with self._handle as capsule, self._lock:
+        with self._lock:
             if domain_id is not None and domain_id < 0:
                 raise RuntimeError(
                     'Domain id ({}) should not be lower than zero.'
                     .format(domain_id))
-            rclpy_implementation.rclpy_init(
+            self.__context = rclpy_implementation.Context(
```

Comment:
I think this should check if `self.__context` already exists, and error if that's the case. If we don't do that then it's possible to initialize two contexts such that the first context lives until it's garbage collected.